Sources/Sentry/SentryClient.m

```diff
@@ -469,15 +469,18 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
         BOOL shouldAttachStacktrace = alwaysAttachStacktrace || self.options.attachStacktrace
             || (nil != event.exceptions && [event.exceptions count] > 0);
 
-        BOOL debugMetaNotAttached = !(nil != event.debugMeta && event.debugMeta.count > 0);
-        if (!isCrashEvent && shouldAttachStacktrace && debugMetaNotAttached) {
-            event.debugMeta = [self.debugImageProvider getDebugImages];
-        }
-
         BOOL threadsNotAttached = !(nil != event.threads && event.threads.count > 0);
         if (!isCrashEvent && shouldAttachStacktrace && threadsNotAttached) {
             event.threads = [self.threadInspector getCurrentThreads];
         }
+
+        BOOL debugMetaNotAttached = !(nil != event.debugMeta && event.debugMeta.count > 0);
+        if (!isCrashEvent && shouldAttachStacktrace && debugMetaNotAttached) {
+            if (event.threads == nil)
+                event.debugMeta = [self.debugImageProvider getDebugImages];
```

Comment:
`m`: If the event doesn't have any threads, which shouldn't happen, I think we can skip adding the debug images. I think we should add a test for this.

Reply:
Now that you mentioned, I just checked and there is no way the thread array is nil, maybe empty.
I think is ok to have an empty array for debugMeta when the thread array is also empty, this way both data reflects, and we already have test for that.